### PR TITLE
Query: Include compiler: First part of recursive compiler

### PR DESCRIPTION
**Pull request by anpete**
Handle any Include path having at most one collection include anywhere in the path.

### CODE REVIEW

File: src/EFCore.Relational/Query/Expressions/SelectExpression.cs [hunk 1]
```diff
@@ -205,6 +207,7 @@ public virtual Expression Offset
                     && value != null)
                 {
                     PushDownSubquery();
+                    ExplodeStarProjection();
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Expressions/SelectExpression.cs]: Can you also file issue for me to clean this up?

File: .gitignore [hunk 2]
```diff
@@ -12,6 +12,7 @@
 *.userosscache
 *.sln.docstates
 *.user.sln*
+test/test.cmd
```
Comment (smitpatel) [inline .gitignore]: netcore for win :trollface:

File: src/EFCore.Specification.Tests/QueryTestBase.cs [hunk 3]
```diff
@@ -52,7 +52,7 @@ orderby _c1
             }
         }
 
-        [ConditionalFact(Skip = "Bug in projection shaper. See Issue #8095")]
+        [ConditionalFact]
```
Comment (smitpatel) [inline src/EFCore.Specification.Tests/QueryTestBase.cs]: This got fixed?

Comment (anpete) [inline src/EFCore.Specification.Tests/QueryTestBase.cs]: No, it doesn't need to be disable for in-memory.

**Comment by anpete** [comment]
ping @smitpatel @maumar

File: src/EFCore.Relational/Query/Expressions/SelectExpression.cs [hunk 4]
```diff
@@ -807,7 +810,32 @@ public virtual Ordering AddToOrderBy([NotNull] Ordering ordering)
...
+                            {
+                                return false;
+                            }
+
+                            if (_expressionEqualityComparer.Equals(o.Expression, ordering.Expression))
+                            {
+                                return true;
+                            }
+
+                            if (o.Expression is NullableExpression nullableExpression1
```
Comment (smitpatel) [inline src/EFCore.Relational/Query/Expressions/SelectExpression.cs]: :wow:

**Comment by smitpatel** [comment]
LGTM

File: src/EFCore.Relational/Query/Expressions/SelectExpression.cs (hunk 4 above)
Comment (anpete) [inline src/EFCore.Relational/Query/Expressions/SelectExpression.cs]: Yeah, we should improve this.

**Pull request merged**